python/cudf/cudf/utils/utils.py

```diff
@@ -524,3 +524,25 @@ def _gather_map_is_valid(
         return True
     gm_min, gm_max = minmax(gather_map)
     return gm_min >= -nrows and gm_max < nrows
+
+
+def _make_column_name(
+    name: Any,
+    multilevel: bool,
+    col_level: int = 0,
+    col_fill: Any = "",
+    total_level: int = 0,
+):
+    """Make a cudf column name from `name`.
+
+    If `multilevel` is True, return a tuple with len == `total_level`. `name`
+    is the `col_level`th item, the rest is `col_fill`.
+
+    Otherwise, return ``name``.
+    """
+    # Should this be a ColumnAccessor method instead?
```

Comment:
?

Reply:
Oh gosh. Maybe? Is there anywhere other than `reset_index` that needs to be doing this?

Reply:
Not that I know of for now. Creating a column name for column accessors with multi levels sounds a common use case and very much dependent on the object. This is where it came from.

Reply:
I would suggest trying to find other use cases in the code right now. If there aren't any, I would probably just inline this logic for now and come back to it later. I'm wary of anticipating when a function will be reusable, we have too many single-use functions floating around as a result of that premature optimization (and some functions that go completely unused because the call site is removed and the function is orphaned).